src/github.com/couchbaselabs/sync_gateway/base/bucket.go

```diff
@@ -105,6 +130,123 @@ func (bucket couchbaseBucket) StartTapFeed(args walrus.TapArguments) (walrus.Tap
 	return &impl, nil
 }
 
+// Start cbdatasource-based DCP feed.  Supports
+func (bucket couchbaseBucket) StartDCPFeed(args walrus.TapArguments) (walrus.TapFeed, error) {
+
+	// Recommended usage of cbdatasource is to let it manage it's own dedicated connection, so we're not
+	// reusing the bucket connection we've already established.
+	urls := []string{bucket.spec.Server}
+	poolName := bucket.spec.PoolName
+	if poolName == "" {
+		poolName = "default"
+	}
+	bucketName := bucket.spec.BucketName
+
+	vbucketIdsArr := []uint16(nil) // nil means get all the vbuckets.
+
+	receiver := NewDCPReceiver()
+
+	maxVbno, err := bucket.getMaxVbno()
+	if err != nil {
+		return nil, err
+	}
+
+	startSeqnos := make(map[uint16]uint64, maxVbno)
+	vbuuids := make(map[uint16]uint64, maxVbno)
+
+	// GetStatsVbSeqno retrieves high sequence number for each vbucket, to enable starting
+	// DCP stream from that position.  Also being used as a check on whether the server supports
+	// DCP.
```

Comment:
I think the above comment might be better placed above the `bucket.GetStatsVbSeqno()` method definition.